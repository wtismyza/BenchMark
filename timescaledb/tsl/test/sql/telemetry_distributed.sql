-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;

SET client_min_messages TO ERROR;
DROP DATABASE IF EXISTS data_node_1;
DROP DATABASE IF EXISTS data_node_2;
SET client_min_messages TO NOTICE;

SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');

-- Become an access node
SELECT * FROM add_data_node('data_node_1', host => 'localhost',
                            database => 'data_node_1');

SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');

-- See telemetry report from a data node
\c data_node_1 :ROLE_CLUSTER_SUPERUSER;
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');

\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
SELECT * FROM add_data_node('data_node_2', 'localhost',
                            database => 'data_node_2');

-- Add hypertables 
CREATE TABLE test_ht(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_hypertable('test_ht', 'time', 'device', 1);

CREATE TABLE disttable(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_distributed_hypertable('disttable', 'time', 'device', 2);

CREATE TABLE disttable2(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_distributed_hypertable('disttable2', 'time', 'device', 2, replication_factor => 2);

-- See telemetry report update from the data node
\c data_node_1 :ROLE_CLUSTER_SUPERUSER;
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');

\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
-- See a number of distributed and distributed and replicated hypertables update
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
