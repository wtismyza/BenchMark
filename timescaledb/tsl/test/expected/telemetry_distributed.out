-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.
\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
SET client_min_messages TO ERROR;
DROP DATABASE IF EXISTS data_node_1;
DROP DATABASE IF EXISTS data_node_2;
SET client_min_messages TO NOTICE;
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
       json_object_field        
--------------------------------
 {"distributed_member": "none"}
(1 row)

-- Become an access node
SELECT * FROM add_data_node('data_node_1', host => 'localhost',
                            database => 'data_node_1');
  node_name  |   host    | port  |  database   | node_created | database_created | extension_created 
-------------+-----------+-------+-------------+--------------+------------------+-------------------
 data_node_1 | localhost | 55432 | data_node_1 | t            | t                | t
(1 row)

SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
                                                                                      json_object_field                                                                                      
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"num_data_nodes": "1", "distributed_member": "access node", "num_distributed_hypertables": "0", "num_distributed_hypertables_members": "0", "num_replicated_distributed_hypertables": "0"}
(1 row)

-- See telemetry report from a data node
\c data_node_1 :ROLE_CLUSTER_SUPERUSER;
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
                                                                                     json_object_field                                                                                     
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"num_data_nodes": "0", "distributed_member": "data node", "num_distributed_hypertables": "0", "num_distributed_hypertables_members": "0", "num_replicated_distributed_hypertables": "0"}
(1 row)

\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
SELECT * FROM add_data_node('data_node_2', 'localhost',
                            database => 'data_node_2');
  node_name  |   host    | port  |  database   | node_created | database_created | extension_created 
-------------+-----------+-------+-------------+--------------+------------------+-------------------
 data_node_2 | localhost | 55432 | data_node_2 | t            | t                | t
(1 row)

-- Add hypertables 
CREATE TABLE test_ht(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_hypertable('test_ht', 'time', 'device', 1);
 hypertable_id | schema_name | table_name | created 
---------------+-------------+------------+---------
             1 | public      | test_ht    | t
(1 row)

CREATE TABLE disttable(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_distributed_hypertable('disttable', 'time', 'device', 2);
 hypertable_id | schema_name | table_name | created 
---------------+-------------+------------+---------
             2 | public      | disttable  | t
(1 row)

CREATE TABLE disttable2(time timestamptz, device int, PRIMARY KEY (time, device));
SELECT * FROM create_distributed_hypertable('disttable2', 'time', 'device', 2, replication_factor => 2);
 hypertable_id | schema_name | table_name | created 
---------------+-------------+------------+---------
             3 | public      | disttable2 | t
(1 row)

-- See telemetry report update from the data node
\c data_node_1 :ROLE_CLUSTER_SUPERUSER;
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
                                                                                     json_object_field                                                                                     
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"num_data_nodes": "0", "distributed_member": "data node", "num_distributed_hypertables": "0", "num_distributed_hypertables_members": "2", "num_replicated_distributed_hypertables": "0"}
(1 row)

\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
-- See a number of distributed and distributed and replicated hypertables update
SELECT json_object_field(get_telemetry_report(always_display_report := true)::json, 'distributed_db');
                                                                                      json_object_field                                                                                      
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"num_data_nodes": "2", "distributed_member": "access node", "num_distributed_hypertables": "2", "num_distributed_hypertables_members": "0", "num_replicated_distributed_hypertables": "1"}
(1 row)

